plugins/amd-kria/fu-amd-kria-device.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2018 Richard Hughes <richard@hughsie.com>
+ * Copyright 2024 Advanced Micro Devices Inc.
+ *
+ * This file is provided under a dual MIT/LGPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ * AMD Chooses the MIT license part of Dual MIT/LGPLv2 license agreement.
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR MIT
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-amd-kria-device.h"
+
+struct _FuAmdKriaDevice {
+	FuDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuAmdKriaDevice, fu_amd_kria_device, FU_TYPE_DEVICE)
+
+static gboolean
+fu_amd_kria_device_write_firmware(FuDevice *device,
+				  FuFirmware *firmware,
+				  FuProgress *progress,
+				  FwupdInstallFlags flags,
+				  GError **error)
+{
+	FuContext *ctx = fu_device_get_context(device);
+	g_autofree gchar *cod_path = NULL;
+	g_autoptr(GBytes) fw = NULL;
+
+	fw = fu_firmware_get_bytes(firmware, error);
+	if (fw == NULL)
+		return FALSE;
+	cod_path = g_build_filename(fu_context_get_esp_location(ctx),
+				    "EFI", "UpdateCapsule", "fwupd.cap", NULL);
```

Comment:
`fwupd.cap` -- what happens if we schedule more than one update? or is there only one item in the ESRT table?

Reply:
It's specifically only for a "system firmware" solution for the SOM.  Yes; there is only one entry that corresponds to the image partition on the QSPI.

BTW - in testing I noticed I'm missing a prepare vfunc to mount the ESP 🤦.  I'll amend.